stratum/lib/security/credentials_manager.cc

```diff
@@ -1,130 +1,161 @@
 // Copyright 2018 Google LLC
 // Copyright 2018-present Open Networking Foundation
+// Copyright 2023 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 
 #include "stratum/lib/security/credentials_manager.h"
 
 #include <memory>
 #include <string>
 #include <utility>
+#include <fstream>
+#include <iostream>
 
 #include "absl/memory/memory.h"
+#include "absl/time/clock.h"
+#include "absl/time/time.h"
 #include "gflags/gflags.h"
-#include "grpcpp/security/server_credentials.h"
-#include "grpcpp/security/tls_credentials_options.h"
 #include "stratum/glue/logging.h"
-#include "stratum/glue/status/status.h"
 #include "stratum/lib/macros.h"
 #include "stratum/lib/utils.h"
 
-DEFINE_string(ca_cert, "", "CA certificate path");
-DEFINE_string(server_key, "", "gRPC Server pricate key path");
-DEFINE_string(server_cert, "", "gRPC Server certificate path");
+DEFINE_string(ca_cert_file, "", "Path to CA certificate file");
+DEFINE_string(server_key_file, "", "Path to gRPC server private key file");
+DEFINE_string(server_cert_file, "", "Path to gRPC server certificate file");
+DEFINE_string(client_key_file, "", "Path to gRPC client key file");
+DEFINE_string(client_cert_file, "", "Path to gRPC client certificate file");
 
 namespace stratum {
 
-CredentialsManager::~CredentialsManager() {}
+using ::grpc::experimental::FileWatcherCertificateProvider;
+using ::grpc::experimental::TlsChannelCredentialsOptions;
+using ::grpc::experimental::TlsServerCredentials;
+using ::grpc::experimental::TlsServerCredentialsOptions;
+
+constexpr unsigned int CredentialsManager::kFileRefreshIntervalSeconds;
+
 CredentialsManager::CredentialsManager() {}
 
+CredentialsManager::~CredentialsManager() {}
+
+::util::StatusOr<std::unique_ptr<CredentialsManager>>
+CredentialsManager::CreateInstance(bool secure_only /*=false*/) {
```

Comment:
Introduced by me: A `secure_only` flag that defaults to `false` to maintain backward compatibility